### PR TITLE
Update dev_env_mac.md

### PR DESCRIPTION
**Pull request by unipheas**
Updated dependency requirements needed for macOS

### CODE REVIEW

File: en/setup/dev_env_mac.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ Install pip if you don't already have it and use it to install the required pack
 
 ```sh
 sudo easy_install pip
-sudo -H pip install pyserial empy toml numpy pandas jinja2 pyyaml
+sudo -H pip install nose tornado pyserial empy toml pandas jinja2 pyyaml
```
Comment (hamishwillee) [inline en/setup/dev_env_mac.md:48]: Hi @unipheas ,
I know nothing about mac. We should use whatever libraries are used in the docker mac environments. Are nose and tornado what they use?

Comment (unipheas) [reply en/setup/dev_env_mac.md:48]: After using a clean MacBook (I wiped mine) I had to install `nose` and `tornado` as they where required by other packages to run. I'm not sure about what's in the docker Mac environment. With the most recent updates, this could have been taken care of.

I'm planning on doing another wipe the beginning of next month. I can go through this process again to make sure everything works from scratch. I may need to get myself another MacBook for things like this 😛

Comment (hamishwillee) [reply en/setup/dev_env_mac.md:48]: Thanks @unipheas . I thought that dependencies like that were automatically pulled in? Anyway I am loath to make this change unless it matches CI.

@julianoes Do you use macOS?

Comment (julianoes) [reply en/setup/dev_env_mac.md:48]: > @julianoes Do you use macOS?

Not unless I have to.

I never installed these two. What were they required for?

Comment (unipheas) [reply en/setup/dev_env_mac.md:48]: When I first setup these were required/missing packages by other packages that are being installed. I think it has something to do with empty or toml. I can't remember without going through the entire process again. 

I'm trying to make it easier for the user to setup instead of having to debug all the errors because they haven't ever setup a development environment or have something else setup.

Comment (hamishwillee) [reply en/setup/dev_env_mac.md:48]: That is very reasonable, but they should not be needed; ie packages for numpy, toml are supposed to pick up their own dependencies. We need someone with a couple of recent clean macOS to test this.

Comment (unipheas) [reply en/setup/dev_env_mac.md:48]: In a few weeks I will completely wipe my MacBook and can try again.

Comment (hamishwillee) [reply en/setup/dev_env_mac.md:48]: Really appreciated. It is possible that what worked in previous versions might have changed, but either way we'd need to establish where the problems lie.

Comment (unipheas) [reply en/setup/dev_env_mac.md:48]: No problem. Jinger and I are doing a workshop today so we'll see what kind of issues people have on fresh installs. That said, I'm having one of the organizers go through the process and they are running into problems already because of Python. Once I get more information then I'll update here with what's going on.

**Comment by unipheas** [comment]
I've wiped my computer and have tried installing everything using [these instructions](https://dev.px4.io/master/en/setup/dev_env_mac.html). I have not installed anything except for Xcode and Homebrew (git & node)

When running `brew install px4-dev` I was presented with this dialog

Screen Shot 2019-12-08 at 12 52 25

In order to get past this I had to go into `System Preferences` > `Security & Privacy` > `General` and click `Open Anyway` at the bottom.

Once I continued I ran into this error
```
ERROR: matplotlib 1.3.1 requires nose, which is not installed.
ERROR: matplotlib 1.3.1 requires tornado, which is not installed.
ERROR: pandas 0.24.2 has requirement numpy>=1.12.0, but you'll have numpy 1.8.0rc1 which is incompatible.
Installing collected packages: pyserial, empy, toml, python-dateutil, pandas, MarkupSafe, jinja2, pyyaml
  Found existing installation: python-dateutil 1.5
    Uninstalling python-dateutil-1.5:
ERROR: Could not install packages due to an EnvironmentError: [('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/parser.pyc', '/private/tmp/pip-uninstall-Whuv7A/parser.pyc', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/parser.pyc'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/relativedelta.py', '/private/tmp/pip-uninstall-Whuv7A/relativedelta.py', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/relativedelta.py'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/zoneinfo/zoneinfo-2010g.tar.gz', '/private/tmp/pip-uninstall-Whuv7A/zoneinfo/zoneinfo-2010g.tar.gz', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/zoneinfo/zoneinfo-2010g.tar.gz'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/zoneinfo/__init__.py', '/private/tmp/pip-uninstall-Whuv7A/zoneinfo/__init__.py', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/zoneinfo/__init__.py'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/zoneinfo/__init__.pyc', '/private/tmp/pip-uninstall-Whuv7A/zoneinfo/__init__.pyc', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/zoneinfo/__init__.pyc'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/zoneinfo', '/private/tmp/pip-uninstall-Whuv7A/zoneinfo', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/zoneinfo'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/tz.py', '/private/tmp/pip-uninstall-Whuv7A/tz.py', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/tz.py'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/relativedelta.pyc', '/private/tmp/pip-uninstall-Whuv7A/relativedelta.pyc', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/relativedelta.pyc'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/rrule.pyc', '/private/tmp/pip-uninstall-Whuv7A/rrule.pyc', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/rrule.pyc'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/__init__.py', '/private/tmp/pip-uninstall-Whuv7A/__init__.py', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/__init__.py'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/parser.py', '/private/tmp/pip-uninstall-Whuv7A/parser.py', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/parser.py'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/tzwin.py', '/private/tmp/pip-uninstall-Whuv7A/tzwin.py', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/tzwin.py'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/rrule.py', '/private/tmp/pip-uninstall-Whuv7A/rrule.py', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/rrule.py'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/__init__.pyc', '/private/tmp/pip-uninstall-Whuv7A/__init__.pyc', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/__init__.pyc'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/easter.py', '/private/tmp/pip-uninstall-Whuv7A/easter.py', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/easter.py'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/easter.pyc', '/private/tmp/pip-uninstall-Whuv7A/easter.pyc', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/easter.pyc'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/tz.pyc', '/private/tmp/pip-uninstall-Whuv7A/tz.pyc', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/tz.pyc'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil/tzwin.pyc', '/private/tmp/pip-uninstall-Whuv7A/tzwin.pyc', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A/tzwin.pyc'"), ('/System/Library/Frameworks/Python.framework/Versions/2.7/Extras/lib/python/dateutil', '/private/tmp/pip-uninstall-Whuv7A', "[Errno 1] Operation not permitted: '/private/tmp/pip-uninstall-Whuv7A'")]
```

I then had to install `nose` and `tornado`. Even at this point it didn't work because of numpy. I have given up at this point because I'm not familiar with `python/pip` and I don't want to spend hours debugging everything.

Here is the [full log](https://gist.github.com/unipheas/85efd28a94777d2320c758837f8a9d1e). You can see the difficulty I had line by line trying to install this.
Basically I didn't get past `sudo -H pip install pyserial empy toml numpy pandas jinja2 pyyaml`

**Comment by unipheas** [comment]
I tried to run the simulator and this was the result. I hope you can see how difficult this can be for a beginner trying to get into this.

```
√ ~ % cd Developer/Firmware 
√ Firmware % make px4_sitl jmavsim                                                                   (master)Firmware
-- PX4 version: v1.10.0-beta4-470-g0a59d8a36d
-- PX4 config file: /Users/brianphillips/Developer/Firmware/boards/px4/sitl/default.cmake
-- PX4 config: px4_sitl_default
-- PX4 platform: posix
-- PX4 lockstep: enabled
-- cmake build type: RelWithDebInfo
-- The CXX compiler identification is AppleClang 11.0.0.11000033
-- The C compiler identification is AppleClang 11.0.0.11000033
-- The ASM compiler identification is Clang
-- Found assembler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Found PythonInterp: /usr/bin/python (found version "2.7.16") 
CMake Error at /usr/local/Cellar/cmake/3.15.5/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  couldn't find python module jinja2:

  		

  for debian systems try: 		

  	sudo apt-get install python-jinja2 		

  or for all other OSs/debian: 		

  	sudo -H pip install jinja2

   (missing: PY_JINJA2) 
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.15.5/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  cmake/px4_find_python_module.cmake:67 (find_package_handle_standard_args)
  CMakeLists.txt:277 (px4_find_python_module)


-- Configuring incomplete, errors occurred!
See also "/Users/brianphillips/Developer/Firmware/build/px4_sitl_default/CMakeFiles/CMakeOutput.log".
See also "/Users/brianphillips/Developer/Firmware/build/px4_sitl_default/CMakeFiles/CMakeError.log".
Error: /Users/brianphillips/Developer/Firmware/build/px4_sitl_default is not a directory
make: *** [px4_sitl] Error 1
?2 Firmware %   
```

**Comment by hamishwillee** [comment]
@unipheas Thanks. 

@julianoes How do you want to progress this? Obviously we should add anything needed, but I'd like to make sure that what we do matches with what docker does. Who else can we ask?

**Comment by julianoes** [comment]
Hm odd. openjdk 13 won't help anyway because you need Java 8 for jMAVSim.

**Comment by unipheas** [comment]
Just a quick update. I had to run the following to get the simulator to work

```
sudo -H pip install jinja2
sudo pip install pyyaml
```

Then I had to go install Java 8 JDK for everything to work

**Comment by hamishwillee** [comment]
Thanks. FWIW the pymaml and jinja2 should already be present if following the instructions:
```
 sudo -H pip install pyserial empy toml numpy pandas jinja2 pyyaml
```

**Comment by unipheas** [comment]
> Thanks. FWIW the pymaml and jinja2 should already be present if following the instructions:
> 
> ```
>  sudo -H pip install pyserial empy toml numpy pandas jinja2 pyyaml
> ```

Yea, I think the problem is that those don't install if a previous package fails to install. Anyway, those are the steps I took to install on a clean MacBook. Do we want to add these two package and close this now?

**Comment by hamishwillee** [comment]
I'm deferring to @julianoes . Personally I do not want to add them; I want to understand why docker gets away without them.

But if we can't get information on that, then a compromise might be to add these two new ones in a separate line with a comment or in a note stating that they are anecdotally needed and we're still checking.

**Comment by julianoes** [comment]
@unipheas I don't think I agree with how you install Python and pip, and I have the feeling that's where it gets messy.

```
√ ~ % sudo easy_install pip
Password:
Searching for pip
Reading https://pypi.org/simple/pip/
Downloading https://files.pythonhosted.org/packages/00/b6/9cfa56b4081ad13874b0c6f96af8ce16cfbc1cb06bedf8e9164ce5551ec1/pip-19.3.1-py2.py3-none-any.whl#sha256=6917c65fc3769ecdc61405d3dfd97afdedd75808d200b2838d7d961cebc0c2c7
Best match: pip 19.3.1
Processing pip-19.3.1-py2.py3-none-any.whl
Installing pip-19.3.1-py2.py3-none-any.whl to /Library/Python/2.7/site-packages
Adding pip 19.3.1 to easy-install.pth file
Installing pip script to /usr/local/bin
Installing pip3.7 script to /usr/local/bin
Installing pip3 script to /usr/local/bin
```

Can you paste the output of:

```
which python
python --version
which pip
pip --version
which python3
python3 --version
which pip3
pip3 --version
```

**Comment by unipheas** [comment]
> @unipheas I don't think I agree with how you install Python and pip, and I have the feeling that's where it gets messy.
> 
> ```
> √ ~ % sudo easy_install pip
> Password:
> Searching for pip
> Reading https://pypi.org/simple/pip/
> Downloading https://files.pythonhosted.org/packages/00/b6/9cfa56b4081ad13874b0c6f96af8ce16cfbc1cb06bedf8e9164ce5551ec1/pip-19.3.1-py2.py3-none-any.whl#sha256=6917c65fc3769ecdc61405d3dfd97afdedd75808d200b2838d7d961cebc0c2c7
> Best match: pip 19.3.1
> Processing pip-19.3.1-py2.py3-none-any.whl
> Installing pip-19.3.1-py2.py3-none-any.whl to /Library/Python/2.7/site-packages
> Adding pip 19.3.1 to easy-install.pth file
> Installing pip script to /usr/local/bin
> Installing pip3.7 script to /usr/local/bin
> Installing pip3 script to /usr/local/bin
> ```
> 
> Can you paste the output of:
> 
> ```
> which python
> python --version
> which pip
> pip --version
> which python3
> python3 --version
> which pip3
> pip3 --version
> ```

```
√ ~ % which python
/usr/bin/python
√ ~ % python --version
Python 2.7.16
√ ~ % which pip
/usr/local/bin/pip
√ ~ % pip --version
pip 19.3.1 from /Library/Python/2.7/site-packages/pip-19.3.1-py2.7.egg/pip (python 2.7)
√ ~ % which python3 
/usr/local/bin/python3
√ ~ % python3 --version
Python 3.7.5
√ ~ % which pip3
/usr/local/bin/pip3
√ ~ % pip3 --version
pip 19.3.1 from /usr/local/lib/python3.7/site-packages/pip (python 3.7)
```

I'd like to note that I was following the instructions listed above, but they seem to have changed now. This issue may be obsolete with that in mind. I also see you've making a new fix for macOS #922. I think I'd agree with you on how I'm installing everything too.

Maybe we can close this now?

**Comment by hamishwillee** [comment]
Closing. https://github.com/PX4/Devguide/pull/922 changes things enough that we should create a new issue if there are ongoing problems and cross link to this.
